apps/advisor/pkg/app/checks/datasourcecheck/prom_dep_auth_check_step.go

```diff
@@ -0,0 +1,162 @@
+	}
+	if dataSource.Type != datasources.DS_PROMETHEUS {
+		return nil, nil
+	}
+	if dataSource.JsonData == nil {
+		return nil, nil
+	}
+
+	awsAuthLinks, err := s.checkUsingAWSAuth(ctx, dataSource)
+	if err != nil {
+		return nil, err
+	}
+	azureAuthLinks, err := s.checkUsingAzureAuth(ctx, dataSource)
+	if err != nil {
+		return nil, err
+	}
+
+	errorLinks := append(awsAuthLinks, azureAuthLinks...)
+
+	if len(errorLinks) == 0 {
+		return nil, nil
+	}
+
+	return []advisor.CheckReportFailure{checks.NewCheckReportFailureWithMoreInfo(
+		advisor.CheckReportFailureSeverityHigh,
+		s.ID(),
+		dataSource.Name,
+		dataSource.UID,
+		errorLinks,
+		fmt.Sprintf("Plugin: %s", dataSource.Type),
+	)}, nil
+}
+
+func (s *promDepAuthStep) checkUsingAWSAuth(ctx context.Context, dataSource *datasources.DataSource) ([]advisor.CheckErrorLink, error) {
+	var errorLinks []advisor.CheckErrorLink
+	if sigV4Auth, found := dataSource.JsonData.CheckGet("sigV4Auth"); found {
+		if enabled, err := sigV4Auth.Bool(); err != nil || !enabled {
+			return nil, err
+		}
+		readOnlyLink, err := checkReadOnly(dataSource)
+		if err != nil {
+			return nil, err
+		}
+
+		if readOnlyLink != nil {
+			errorLinks = append(errorLinks, *readOnlyLink)
+		}
+
+		errorLinks = append(errorLinks,
+			advisor.CheckErrorLink{
+				Message: "View SigV4 docs",
+				Url:     "https://grafana.com/docs/grafana-cloud/connect-externally-hosted/data-sources/prometheus/configure/aws-authentication/",
+			})
+		pluginLink, err := s.linkDataSource(ctx, datasources.DS_AMAZON_PROMETHEUS)
+		if err != nil {
+			return nil, err
+		}
+		if pluginLink != nil {
+			errorLinks = append(errorLinks, *pluginLink)
+		}
+	}
+	return errorLinks, nil
+}
+
+func (s *promDepAuthStep) checkUsingAzureAuth(ctx context.Context, dataSource *datasources.DataSource) ([]advisor.CheckErrorLink, error) {
+	var errorLinks []advisor.CheckErrorLink
+	if azureAuth, found := dataSource.JsonData.CheckGet("azureCredentials"); found {
+		if _, err := azureAuth.Value(); err != nil {
+			return nil, err
+		}
+		readOnlyLink, err := checkReadOnly(dataSource)
+		if err != nil {
+			return nil, err
+		}
+		if readOnlyLink != nil {
+			errorLinks = append(errorLinks, *readOnlyLink)
+		}
+		errorLinks = append(errorLinks,
+			advisor.CheckErrorLink{
+				Message: "View Azure auth docs",
+				Url:     "https://grafana.com/docs/grafana-cloud/connect-externally-hosted/data-sources/prometheus/configure/azure-authentication/",
+			})
+		pluginLink, err := s.linkDataSource(ctx, datasources.DS_AZURE_PROMETHEUS)
+		if err != nil {
+			return errorLinks, err
+		}
+		if pluginLink != nil {
+			errorLinks = append(errorLinks, *pluginLink)
+		}
+	}
+	return errorLinks, nil
+}
+
+func checkReadOnly(dataSource *datasources.DataSource) (*advisor.CheckErrorLink, error) {
+	if readOnly, found := dataSource.JsonData.CheckGet("readonly"); found {
+		if enabled, err := readOnly.Bool(); err != nil || !enabled {
+			return nil, err
```

Comment:
same here, avoid the error, it would mean that readonly is not a boolean and therefore is not readonly.